### PR TITLE
runtime: Fix configmap/secrets updates with FS sharing disabled

### PR DESCRIPTION
**Pull request by Sumynwa**
This PR fixes k8's configmap/secrets etc update propagation when filesystem sharing is disabled.
The commit introduces below changes with some limitations:
- creates new timestamped directory in guest
- updates the '..data' symlink
- creates user visible symlinks to newly created secrets.
- Limitation: The older timestamped directory and stale user visible symlinks exist in guest due to missing DELETE api in agent.

Fixes: #7398

### CODE REVIEW

File: src/runtime/virtcontainers/fs_share_linux.go
Comment (fidencio) [inline src/runtime/virtcontainers/fs_share_linux.go:36]: `// as recommended in review comments for PR #7211` -- I'd personally not add this, but as it's I wouldn't remove either. :-)

**Review by fidencio — COMMENTED** [review]
Code-wise it looks good to me.

You have my "lgtm", and will have my approval once someone can have this one tested on the peer-pods side.  Maybe @stevenhorsman, in case he has a handy cluster around.

**Comment by stevenhorsman** [comment]
> You have my "lgtm", and will have my approval once someone can have this one tested on the peer-pods side. Maybe @stevenhorsman, in case he has a handy cluster around.

So the issue is that peer pods can't run against main here. I did create https://github.com/kata-containers/kata-containers/issues/8015 to cover adding tests for this scenario, but I don't want to block this change waiting for that.

File: src/runtime/virtcontainers/fs_share_linux.go
Comment (bpradipt) [inline src/runtime/virtcontainers/fs_share_linux.go:44]: nit: Either this comment should be removed or the following additional commit added
// `^/var/lib/kubelet/pods/[a-fA-F0-9\-]+/volumes/kubernetes\.io~(configmap|secret|projected|downward-api)`

Comment (Sumynwa) [reply src/runtime/virtcontainers/fs_share_linux.go]: I have removed the comment. Thankyou @bpradipt .

**Comment by danmihai1** [comment]
Sumedh is out on vacation. I will remind him to address the feedback from this PR when he comes back to work, next week.

**Comment by danmihai1** [comment]
> Sumedh is out on vacation. I will remind him to address the feedback from this PR when he comes back to work, next week.

Nevermind - looks like he is back already.

**Comment by Sumynwa** [comment]
Dan, I have addressed the PR feedbacks.

**Review by fidencio — APPROVED** [review]
lgtm, thanks @Sumynwa!

**Review by danmihai1 — APPROVED** [review]

**Pull request merged**
